statsd/aggregator.go

```diff
@@ -172,12 +172,34 @@ func (a *aggregator) flushMetrics() []metric {
 }
 
 func getContext(name string, tags []string) string {
-	return name + ":" + strings.Join(tags, tagSeparatorSymbol)
+	c, _ := getContextAndTags(name, tags)
+	return c
 }
 
 func getContextAndTags(name string, tags []string) (string, string) {
-	stringTags := strings.Join(tags, tagSeparatorSymbol)
-	return name + ":" + stringTags, stringTags
+	if len(tags) == 0 {
+		return name + nameSeparatorSymbol, ""
+	}
+	n := len(name) + len(nameSeparatorSymbol) + len(tagSeparatorSymbol)*(len(tags)-1)
+	for _, s := range tags {
+		n += len(s)
+	}
+
+	var sb strings.Builder
+	sb.Grow(n)
+	sb.WriteString(name)
+	sb.WriteString(nameSeparatorSymbol)
+	if len(tags) > 0 {
```

Comment:
We already catch this use case in the if at the top now.

Reply:
Oops, missed that. Thanks for noticing! Fixed.